catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloadManager.java

```diff
@@ -49,53 +46,26 @@
     private static final Logger LOGGER =
             LoggerFactory.getLogger(ReliableResourceDownloadManager.class);
 
-    private DownloadsStatusEventPublisher eventPublisher;
-
-    private ResourceResponse resourceResponse;
-
-    private String downloadIdentifier;
-
-    private DownloadStatusInfo downloadStatusInfo;
-
-    private ExecutorService executor;
-
-    private ReliableResourceDownloaderConfig downloaderConfig =
-            new ReliableResourceDownloaderConfig();
+    private ReliableResourceDownloaderConfig downloaderConfig;
 
     /**
-     * @param resourceCache
-     *            reference to the @ResourceCacheImpl to cache the resource in
-     * @param eventPublisher
-     *            reference to the publisher of status events as the download progresses
-     * @param eventListener
-     *            reference to the {@link DownloadsStatusEventListener}
-     * @param downloadStatusInfo
-     *            reference to the {@link DownloadStatusInfo}
+     * @param downloaderConfig
+     *            reference to the {@link ReliableResourceDownloaderConfig}
      */
-    public ReliableResourceDownloadManager(ResourceCacheImpl resourceCache,
-            DownloadsStatusEventPublisher eventPublisher,
-            DownloadsStatusEventListener eventListener, DownloadStatusInfo downloadStatusInfo,
-            ExecutorService executor) {
-        this.downloaderConfig.setResourceCache(resourceCache);
-        this.eventPublisher = eventPublisher;
-        this.downloaderConfig.setEventPublisher(this.eventPublisher);
-        this.downloaderConfig.setEventListener(eventListener);
-        this.downloadStatusInfo = downloadStatusInfo;
-        this.executor = executor;
-        this.downloadIdentifier = UUID.randomUUID()
-                .toString();
+    public ReliableResourceDownloadManager(ReliableResourceDownloaderConfig downloaderConfig) {
+        this.downloaderConfig = downloaderConfig;
     }
 
     public void init() {
 
     }
 
     public void cleanUp() {
-        executor.shutdown();
+        downloaderConfig.shutdown();
```

Comment:
Feels weird to call a `Config` object to control the `ExecutorService`. See my comment below on the `ReliableResourceDownloaderConfig` class.